cpp/src/io/utilities/hostdevice_vector.hpp

```diff
@@ -190,7 +177,7 @@ class hostdevice_vector {
   }
 
  private:
-  std::variant<thrust::host_vector<T>, cudf::detail::pinned_host_vector<T>> h_data_owner;
+  std::unique_ptr<cudf::detail::rmm_host_vector<T>> h_data_owner;
```

Comment:
Any reason to not just have it be a plain data member? 
```suggestion
 cudf::detail::rmm_host_vector<T> h_data_owner;
```